src/backend/routers/students.ts

```diff
@@ -17,30 +16,85 @@ export const studentProcedures = {
 
       return result;
     }),
+  //* this function gets students for the current logged in case manager
+  getMyStudents: procedure
+    .input(z.object({ assigned_case_manager_id: z.string().uuid() }))
+    .query(async (req) => {
+      const { assigned_case_manager_id } = req.input;
 
-  getAllStudents: procedure.query(async (req) => {
-    const result = await req.ctx.db.selectFrom("student").selectAll().execute();
+      const result = await req.ctx.db
+        .selectFrom("student")
+        .selectAll()
+        .where("assigned_case_manager_id", "=", assigned_case_manager_id)
+        .execute();
 
-    return result;
-  }),
+      return result;
+    }),
 
   createStudent: procedure
     .input(
       z.object({
         first_name: z.string(),
         last_name: z.string(),
         email: z.string(),
+        assigned_case_manager_id: z.string(),
       })
     )
     .mutation(async (req) => {
-      const { first_name, last_name, email } = req.input;
+      const { first_name, last_name, email, assigned_case_manager_id } =
+        req.input;
 
-      const result = await req.ctx.db
-        .insertInto("student")
-        .values({ first_name, last_name, email })
-        .returningAll()
-        .execute();
+      const inDatabase = await req.ctx.db
+        .selectFrom("student")
+        .selectAll()
+        .where("email", "=", email)
+        .executeTakeFirst();
```

Comment:
[`ON CONFLICT`](https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT) can be used in Postgres to perform upserts and reduce the custom logic here.

with kysely I think it's something like `.onConflict(qb => qb.column("email"))...`

Reply:
I'm not sure how the `.onConflict` method can reduce the custom logic when resulting in different error messages. I looked into the other approaches, but still haven't found a good solution.

Reply:
thinking through this a bit more, it seems like there's potentially two separate methods here: one to create a student and one to update / assign a case manager?

Reply:
Yes. The only way I can think of for getting a student's `assigned_case_manager_id` is by looking up that student in the query. So I can keep the `const inDatabase` and utilize `.onConflict` to reduce the custom logic, or is there a better way to do this?

Reply:
the way I would do it is have a single SQL statement built with something like:

```ts
await req.ctx.db
  .insertInto("student")
  .values({ first_name, last_name, email, assigned_case_manager_id })
  .onConflict(qb => qb.column("email").doUpdateSet({
    assigned_case_manager_id
  }).where("student.assigned_case_manager_id", "=", null))
  .execute()
```

this will:
- attempt to insert into `student`
- if there's an existing row with the same `email` value, update the `assigned_case_manager_id` **only if** it's currently null

Reply:
I would also probably rename to something like `createStudentOrAssignManager`

Reply:
That's why I kept getting the alert, `assigned_case_manager_id is ambiguous`!

I made the change, but it doesn't seem to add the student in when the `assigned_case_manager_id = null`. Apparently, Postgres sees the evaluation as [false](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-is-null/). The terminology used in Postgres is `IS NULL`, but I couldn't find its equivalent in tRPC.

Reply:
Actually, I figured it out. I'll push the updated codes.